### ISSUE TITLE
why existed so many bugs with my deploy openresty1.13+modsec-v3+crs-v3.2

### ISSUE THREAD

**Issue by ghost** [SpiderLabs/ModSecurity#1876]
i run modsec3+crs3.02 well but there are so many bug when i deploy the modsecv3 with crs3.2+ .

but when i use modsec3+crs3.0 the dos-attack  run well. 

how can i do it make dos run well with crs3.0.2? 

- this is the issue that i put in `SpiderLabs/owasp-modsecurity-crs` and than them give my this ticket.

[why existed so many bugs with my deploy openresty1.13+modsec-v3+crs-v3.2](https://github.com/SpiderLabs/owasp-modsecurity-crs/issues/1174)

i will rewirte the problem down

**Comment by ghost** [comment]
### errors Show
```
[root@localhost /]# nginx
nginx: [emerg] "modsecurity_rules_file" directive Rules error. File: /etc/nginx/rules/REQUEST-901-INITIALIZATION.conf. Line: 298. Column: 119. Expecting an action, got:  ctl:requestBodyProcessor=URLENCODED" in /etc/nginx/nginx.conf:41
[root@localhost /]# vi /etc/nginx/rules/REQUEST-901-INITIALIZATION.conf 
[root@localhost /]# vi /etc/nginx/rules/REQUEST-901-INITIALIZATION.conf 
[root@localhost /]# nginx
nginx: [emerg] "modsecurity_rules_file" directive Rules error. File: /etc/nginx/rules/REQUEST-912-DOS-PROTECTION.conf. Line: 187. Column: 29. Expecting an action, got:  ,\ in /etc/nginx/nginx.conf:41
[root@localhost /]# vi /etc/nginx/rules/REQUEST-912-DOS-PROTECTION.conf 
[root@localhost /]# 
[root@localhost /]# nginx
nginx: [emerg] "modsecurity_rules_file" directive Rules error. File: /etc/nginx/rules/REQUEST-914-FILE-DETECTION.conf. Line: 70. Column: 85. Expecting a variable, got:  :  S}: %{MATCHED_VARS}',\ in /etc/nginx/nginx.conf:41
[root@localhost /]# vi /etc/nginx/rules/REQUEST-914-FILE-DETECTION.conf 
[root@localhost /]# ^C
[root@localhost /]# ^C
[root@localhost /]# vi /etc/nginx/rules/REQUEST-914-FILE-DETECTION.conf 
[root@localhost /]# vi /etc/nginx/rules/REQUEST-914-FILE-DETECTION.conf 
[root@localhost /]# 
[root@localhost /]# nginx
nginx: [emerg] "modsecurity_rules_file" directive Rules error. File: /etc/nginx/rules/REQUEST-914-FILE-DETECTION.conf. Line: 70. Column: 82. Expecting a variable, got:  :  S}: DataReplaced',\ in /etc/nginx/nginx.conf:41
[root@localhost /]# vi /etc/nginx/rules/REQUEST-914-FILE-DETECTION.conf 
[root@localhost /]# nginx
nginx: [emerg] "modsecurity_rules_file" directive Rules error. File: /etc/nginx/rules/REQUEST-914-FILE-DETECTION.conf. Line: 70. Column: 81. Expecting a variable, got:  :  S}: DataReplaed',\ in /etc/nginx/nginx.conf:41

```
### MyNginx 

```
[root@localhost /]# nginx -V
nginx version: openresty/1.13.6.2
built by gcc 4.8.5 20150623 (Red Hat 4.8.5-28) (GCC) 
built with OpenSSL 1.0.2k-fips  26 Jan 2017
TLS SNI support enabled
configure arguments: --prefix=/usr/share/nginx/nginx --with-debug --with-cc-opt='-DNGX_LUA_USE_ASSERT -DNGX_LUA_ABORT_AT_PANIC -O2 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic' --add-module=../ngx_devel_kit-0.3.0 --add-module=../echo-nginx-module-0.61 --add-module=../xss-nginx-module-0.06 --add-module=../ngx_coolkit-0.2rc3 --add-module=../set-misc-nginx-module-0.32 --add-module=../form-input-nginx-module-0.12 --add-module=../encrypted-session-nginx-module-0.08 --add-module=../srcache-nginx-module-0.31 --add-module=../ngx_lua-0.10.13 --add-module=../ngx_lua_upstream-0.07 --add-module=../headers-more-nginx-module-0.33 --add-module=../array-var-nginx-module-0.05 --add-module=../memc-nginx-module-0.19 --add-module=../redis2-nginx-module-0.15 --add-module=../redis-nginx-module-0.3.7 --add-module=../rds-json-nginx-module-0.15 --add-module=../rds-csv-nginx-module-0.09 --add-module=../ngx_stream_lua-0.0.5 --with-ld-opt='-Wl,-rpath,/usr/share/nginx/luajit/lib -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -Wl,-E' --sbin-path=/usr/sbin/nginx --modules-path=/usr/lib64/nginx/modules --conf-path=/etc/nginx/nginx.conf --error-log-path=/var/log/nginx/error.log --http-log-path=/var/log/nginx/access.log --http-client-body-temp-path=/var/lib/nginx/tmp/client_body --http-proxy-temp-path=/var/lib/nginx/tmp/proxy --http-fastcgi-temp-path=/var/lib/nginx/tmp/fastcgi --http-uwsgi-temp-path=/var/lib/nginx/tmp/uwsgi --http-scgi-temp-path=/var/lib/nginx/tmp/scgi --pid-path=/run/nginx.pid --lock-path=/run/lock/subsys/nginx --user=nginx --group=nginx --with-file-aio --with-ipv6 --with-http_auth_request_module --with-http_ssl_module --with-http_v2_module --with-http_realip_module --with-http_addition_module --with-http_xslt_module=dynamic --with-http_image_filter_module=dynamic --with-http_geoip_module=dynamic --with-http_sub_module --with-http_dav_module --with-http_flv_module --with-http_mp4_module --with-http_gunzip_module --with-http_gzip_static_module --with-http_random_index_module --with-http_secure_link_module --with-http_degradation_module --with-http_slice_module --with-http_stub_status_module --with-http_perl_module=dynamic --with-mail=dynamic --with-mail_ssl_module --with-pcre --with-pcre-jit --with-stream=dynamic --with-stream_ssl_module --with-google_perftools_module --add-dynamic-module=/usr/local/src/waf/openresty-1.13.6.2/../ModSecurity-nginx --with-stream --with-stream_ssl_module
```

**Comment by arun81** [comment]
The same is happening in my setup 

Aug 22 07:42:33 ubuntu nginx[1920]: nginx: [emerg] "modsecurity_rules_file" directive Rules error. File: /nginx/modsecurity/rules/REQUEST-901-INITIALIZATION.conf. Line:
Aug 22 07:42:33 ubuntu nginx[1920]: nginx: configuration file /nginx/nginx.conf test failed


root@ubuntu:/nginx/live# service nginx status
● nginx.service - A high performance web server and a reverse proxy server
   Loaded: loaded (/etc/systemd/system/nginx.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Wed 2018-08-22 07:42:33 PDT; 16s ago
  Process: 1906 ExecStop=/sbin/start-stop-daemon --quiet --stop --retry QUIT/5 --pidfile /run/nginx.pid (code=exited, status=0/SUCCESS)
  Process: 1851 ExecStart=/usr/sbin/nginx -g daemon on; master_process on; (code=exited, status=0/SUCCESS)
  Process: 1920 ExecStartPre=/usr/sbin/nginx -t -q -g daemon on; master_process on; (code=exited, status=1/FAILURE)
 Main PID: 1855 (code=exited, status=0/SUCCESS)

Aug 22 07:42:33 ubuntu systemd[1]: Starting A high performance web server and a reverse proxy server...
Aug 22 07:42:33 ubuntu nginx[1920]: nginx: [emerg] "modsecurity_rules_file" directive Rules error. File: /nginx/modsecurity/rules/REQUEST-901-INITIALIZATION.conf. Line:
Aug 22 07:42:33 ubuntu nginx[1920]: nginx: configuration file /nginx/nginx.conf test failed
Aug 22 07:42:33 ubuntu systemd[1]: nginx.service: Control process exited, code=exited status=1
Aug 22 07:42:33 ubuntu systemd[1]: Failed to start A high performance web server and a reverse proxy server.
Aug 22 07:42:33 ubuntu systemd[1]: nginx.service: Unit entered failed state.
Aug 22 07:42:33 ubuntu systemd[1]: nginx.service: Failed with result 'exit-code'.
Aug 22 07:42:36 ubuntu systemd[1]: nginx.service: Unit cannot be reloaded because it is inactive.
Aug 22 07:42:44 ubuntu systemd[1]: nginx.service: Unit cannot be reloaded because it is inactive.
lines 1-17/17 (E

**Comment by AnoopAlias** [comment]
Same error here. I think probably the SpiderLabs/owasp-modsecurity-crs switched default branch to v3.2-dev or something as the v3.0/master branch rules work fine

**Comment by ghost** [comment]
@arun81 you can modify the 901350 urlencode to json then work, but then there will be new bug with 913-dos, then you must delete the 912110 and 912111 these two main rules。。。

you can run with crs3.0, but you can't protect dos-attack...  
[version 3.0.2 dos protection is not work,and version3.0.0 is ok ](https://github.com/SpiderLabs/ModSecurity/issues/1778)

**Comment by ghost** [comment]
@AnoopAlias do you have a try wth dos-attack in crs302 ?

**Comment by ghost** [comment]
I have update my crs-rules v302 to v300, is running,  no  bugs currently ....

[https://codeload.github.com/SpiderLabs/owasp-modsecurity-crs/zip/v3.0.0](https://codeload.github.com/SpiderLabs/owasp-modsecurity-crs/zip/v3.0.0)

**Comment by victorhora** [comment]
Hi,

Yes, https://github.com/SpiderLabs/owasp-modsecurity-crs project changed the default branch from v3.0/master to v3.2/dev (I'm unsure why).

The current stable release of the CRS ([v3.0.2](https://github.com/SpiderLabs/owasp-modsecurity-crs/releases/tag/v3.0.2)) loads cleanly.

The **pre-release** version ([v3.1.0-rc1](https://github.com/SpiderLabs/owasp-modsecurity-crs/releases/tag/v3.1.0-rc1)) should load cleanly since commits https://github.com/SpiderLabs/ModSecurity/commit/f999f54eda0674fe3f453bf939b9abcd2aa0290a and https://github.com/SpiderLabs/ModSecurity/commit/764a2e43ff03370fafa8272f9270160bcad32023 made it through master.

The current development version ([v3.2/dev](https://github.com/SpiderLabs/owasp-modsecurity-crs/tree/v3.2/dev)) should load cleanly as soon as https://github.com/SpiderLabs/owasp-modsecurity-crs/pull/1187 is merged to v3.2/dev branch.
